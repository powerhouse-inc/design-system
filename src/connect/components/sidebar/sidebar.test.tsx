import { render, screen } from '@testing-library/react';

import { ConnectSidebar } from '.';

describe('Connect Sidebar Component', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(
            <ConnectSidebar
                data-testid="sidebar"
                onToggle={() => undefined}
                username="Willow"
                address="0x123"
                collapsed={false}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly', () => {
        render(
            <ConnectSidebar
                data-testid="sidebar"
                onToggle={() => undefined}
                username="Willow"
                address="0x123"
                collapsed={false}
            />,
        );
        expect(screen.getByTestId('sidebar')).toBeInTheDocument();
    });

    it('should render expanded', () => {
        render(
            <ConnectSidebar
                data-testid="sidebar"
                onToggle={() => undefined}
                username="Willow"
                address="0x123"
                collapsed={false}
            />,
        );
        expect(screen.getByTestId('sidebar')).toHaveStyle({ width: '304px' });
    });

    it('should render collapsed', () => {
        render(
            <ConnectSidebar
                data-testid="sidebar"
                onToggle={() => undefined}
                username="Willow"
                address="0x123"
                collapsed={true}
            />,
        );
        expect(screen.getByTestId('sidebar')).toHaveStyle({ width: '58px' });
    });
});
