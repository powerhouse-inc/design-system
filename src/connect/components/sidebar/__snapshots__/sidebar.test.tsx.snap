// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Connect Sidebar Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="group/sidebar group expanded flex h-full flex-col overflow-hidden transition-none duration-300 animate-in fade-in bg-neutral-1 shadow-[0px_33px_32px_-16px_rgba(0,0,0,0.10),0px_0px_16px_4px_rgba(0,0,0,0.04)]"
    data-testid="sidebar"
    style="width: 304px;"
  >
    <div
      class="no-scrollbar flex-1 overflow-auto transition-shadow"
    >
      <div
        class="flex-shrink-0 mb-4 flex justify-center gap-4 px-2 pt-11"
      >
        <input
          class="flex-1 rounded-md border border-neutral-3 px-5 py-3 leading-none placeholder-shown:bg-transparent collapsed:hidden expanding:hidden"
          placeholder="Create new document"
        />
        <button
          class="rounded-md border border-neutral-3 p-3 outline-none hover:bg-slate-50
                collapsed:rotate-180 collapsed:border-[rgba(49,53,56,0.12)] collapsed:bg-neutral-3
                collapsed:shadow-[0px_33px_32px_-16px_rgba(0,0,0,0.10),0px_0px_16px_4px_rgba(0,0,0,0.04)] collapsed:hover:bg-slate-50
                expanding:hidden"
          data-rac=""
          type="button"
        >
          <svg
            style="color: rgb(124, 135, 142); width: 16px; height: 16px;"
          >
            <use
              href="icons.svg#arrow-left"
            />
          </svg>
        </button>
      </div>
      <div
        class="hidden expanded:block"
      />
    </div>
    <div
      class="flex-shrink-0 border-t border-slate-700/10 p-4 collapsed:px-1 expanding:px-1"
    >
      <div
        class="flex gap-2 rounded-sm px-3 py-[10px] collapsed:justify-center
            collapsed:bg-transparent collapsed:px-1 collapsing:bg-transparent
            expanding:justify-center expanding:bg-transparent expanding:px-1
        "
      >
        <img
          alt="Willow"
          class="object-contain"
          height="40"
          src="/src/assets/powerhouse-rounded.png"
          width="40"
        />
        <div
          class="collapsed:hidden expanding:hidden"
        >
          <p
            class="text-gray-800 text-sm font-semibold"
          >
            Willow
          </p>
          <p
            class="text-text-gray-600 text-xs font-semibold"
          >
            0x123
          </p>
        </div>
      </div>
      <button
        class="flex w-full gap-3 px-5 py-3 collapsed:px-3 expanding:px-3"
        data-rac=""
        type="button"
      >
        <svg
          style="width: 24px; height: 24px;"
        >
          <use
            href="icons.svg#settings"
          />
        </svg>
        <span
          class="text-gray-800 text-sm font-semibold leading-6 collapsed:hidden expanding:hidden"
        >
          Settings
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
